include/pybind11/pybind11.h

```diff
@@ -1173,6 +1173,25 @@ class exception : public object {
     }
 };
 
+/** Registers a Python exception in `m` of the given `name` and installs an exception translator to
+ * translate the C++ exception to the created Python exception using the exceptions what() method.
+ * This is intended for simple exception translations; for more complex translation, register the
+ * exception object and translator directly.
+ */
+template <typename CppException> exception<CppException>& register_exception(module &m, const std::string &name, PyObject* base = PyExc_Exception) {
+    static exception<CppException> ex(m, name, base);
+    register_exception_translator([](std::exception_ptr p) {
+        if (!p) return;
+        try {
+            std::rethrow_exception(p);
+        }
+        catch (const CppException &e) {
+            PyErr_SetString(ex.ptr(), e.what());
```

Comment:
Could this be something like `ex.set(e.what())`? This would be useful when writing custom translators too, so you don't have to access raw python api